src/Common/FileCache.h

```diff
@@ -221,7 +222,7 @@ class LRUFileCache final : public IFileCache
     };
 
     using FileSegmentsByOffset = std::map<size_t, FileSegmentCell>;
-    using CachedFiles = std::unordered_map<Key, FileSegmentsByOffset>;
+    using CachedFiles = HashMap<Key, FileSegmentsByOffset, HashCRC32<Key>>;
```

Comment:
Most likely, std::map is not memmove-able type, because the nodes contain a pointer to the parent node.